test/integration/dns_account_01_test.go

```diff
@@ -0,0 +1,383 @@
+//go:build integration
+
+package integration
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/eggsampler/acme/v3"
+)
+
+func TestDNSAccount01HappyPath(t *testing.T) {
+	t.Parallel()
+
+	domain := random_domain()
+	c, err := makeClient()
+	if err != nil {
+		t.Fatalf("creating client: %s", err)
+	}
+
+	idents := []acme.Identifier{{Type: "dns", Value: domain}}
+
+	order, err := c.Client.NewOrder(c.Account, idents)
+	if err != nil {
+		t.Fatalf("creating new order: %s", err)
+	}
+
+	authzURL := order.Authorizations[0]
+	auth, err := c.Client.FetchAuthorization(c.Account, authzURL)
+	if err != nil {
+		t.Fatalf("fetching authorization: %s", err)
+	}
+
+	chal, ok := auth.ChallengeMap[acme.ChallengeTypeDNSAccount01]
+	if !ok {
+		t.Skip("DNS-Account-01 is not offered, skipping test")
+	}
```

Comment:
I think this is a really clever way to condition this test. I think this is a lot cleverer than [our usual `os.Getenv` method](https://github.com/letsencrypt/boulder/blob/fb7906e63debfc7d9e656d5e6674efdc04accd4b/test/integration/email_exporter_test.go#L124).

However, this also makes it very easy for these tests to silently not run at all due to a misconfiguration. I'd almost prefer to replace this with two checks:

1) up near the top:
```go
if os.Getenv("BOULDER_CONFIG_DIR") == "test/config" {
  t.Skip("Test requires dns-account-01 to be enabled")
}
```

2) and then right here:
```go
if !ok {
  t.Fatal("dns-account-01 challenge not offered by server")
}
```